stackslib/src/chainstate/nakamoto/mod.rs

```diff
@@ -362,7 +371,7 @@ impl NakamotoBlockHeader {
     }
 
     pub fn is_first_mined(&self) -> bool {
-        StacksBlockHeader::is_first_index_block_hash(&self.parent_block_id)
+        self.block_id() == StacksBlockId::first_mined()
```

Comment:
This is a change of this method's semantics, right?

That `first_mined()` method returns `StacksBlockId::new(&FIRST_BURNCHAIN_CONSENSUS_HASH, &FIRST_STACKS_BLOCK_HASH)`, which is the _parent_ of the first mined block, right? The first actually mined block should have a real block hash and a real consensus hash.

Reply:
That was a bug, but it was never tripped in my code because my tests only concern themselves with Nakamoto blocks that build atop epoch 2.x blocks (so, code branches that depend on `is_first_mined()` are never used in the first place).